api/src/GTAVersions.Domain/Services/AuthService.cs

```diff
@@ -0,0 +1,59 @@
+﻿using System.Net;
+using System.Threading.Tasks;
+using System.Web.Http;
+using GTAVersions.Domain.DTO;
+using GTAVersions.Domain.Entities;
+using GTAVersions.Domain.Interfaces;
+
+namespace GTAVersions.Domain.Services
+{
+    public class AuthService : IAuthService
+    {
+        private readonly IUserService _userService;
+
+        public AuthService(IUserService userService)
+        {
+            _userService = userService;
+        }
+
+        public async Task<AccessToken> Login(SignInUserDTO request)
+        {
+            var userDTO = await _userService.GetUserByUsernameAsync(request.Username);
+
+            if (userDTO == null)
+            {
+                throw new HttpResponseException(HttpStatusCode.NotFound);
+            }
+
+            if (userDTO.Password != request.Password)
+            {
+                throw new HttpResponseException(HttpStatusCode.Unauthorized);
+            }
+
+            var token = await _userService.UpdateAndReturnUserTokenAsync(userDTO);
+
+            if (token == null)
+            {
+                throw new HttpResponseException(HttpStatusCode.Unauthorized);
+            }
```

Comment:
you can remove this checking. The token can't be null, because there is already checking above that makes sure that we get user/valid user